serde_derive/src/internals/attr.rs

```diff
@@ -1462,6 +1528,88 @@ fn get_lit_str(
     get_lit_str2(cx, attr_name, attr_name, meta)
 }
 
+fn try_get_lit_str(lit: &syn::Expr) -> Result<&syn::LitStr, ()> {
+    if let syn::Expr::Lit(syn::ExprLit {
+        lit: syn::Lit::Str(lit),
+        ..
+    }) = lit
+    {
+        Ok(lit)
+    } else {
+        Err(())
+    }
+}
+
+fn try_get_lit_int(lit: &syn::Expr) -> Result<&syn::LitInt, ()> {
+    if let syn::Expr::Lit(syn::ExprLit {
+        lit: syn::Lit::Int(lit),
+        ..
+    }) = lit
+    {
+        Ok(lit)
+    } else {
+        Err(())
+    }
+}
+
+fn try_get_lit_bool(lit: &syn::Expr) -> Result<&syn::LitBool, ()> {
+    if let syn::Expr::Lit(syn::ExprLit {
+        lit: syn::Lit::Bool(lit),
+        ..
+    }) = lit
+    {
+        Ok(lit)
+    } else {
+        Err(())
+    }
+}
+
+fn get_variant_name(
+    cx: &Ctxt,
+    attr_name: Symbol,
+    meta: &ParseNestedMeta,
+) -> syn::Result<Option<VariantName>> {
+    get_variant_name2(cx, attr_name, attr_name, meta)
+}
+
+// meta_item_name is unused but is present to comply with the contract for
+// get_ser_and_de.
+fn get_variant_name2(
+    cx: &Ctxt,
+    attr_name: Symbol,
+    meta_item_name: Symbol,
+    meta: &ParseNestedMeta,
+) -> syn::Result<Option<VariantName>> {
+    let expr: syn::Expr = meta.value()?.parse()?;
+    let mut value = &expr;
+    while let syn::Expr::Group(e) = value {
+        value = &e.expr;
+    }
+    if let Ok(lit) = try_get_lit_str(value) {
+        Ok(Some(VariantName::String(lit.value())))
```

Comment:
This is missing a check for `lit.suffix().is_empty()`.

It can go in `try_get_lit_str`.